Fix problems after Null-Safety Migration and add some improvements

andresruizdev opened the pull request:
In this pull request, I address the issue found in CurlClient after migrating to Null-Safety and make a series of improvements to the class to support Headers and more request methods using Curl. The main changes made are:

- Explicitly control null values to fix empty values in the args variable after [my last commit](https://github.com/andresruizdev/http_client/commit/27564fbd1b07d5cc0ec5f17dfb5e15c8c892c3ed).
- This PR fixes or closes issue: fixes #13 .
- Add headers support in the curl client.
- Handle disallowed methods according to [HTTP/1.1 (RFC 7231)](https://www.rfc-editor.org/rfc/rfc7231) 
- Update my old username to the new one in the CHANGELOG.md file.
- Update the version (1.5.2).

andresruizdev committed:
Merge remote-tracking branch 'isoos/master'

andresruizdev committed:
Merge branch 'isoos:master' into master

andresruizdev committed:
Fix curl casting error after Null-safety migration

andresruizdev committed:
Version 1.5.2

* Add headers support in curl client.
* Fix Empty Values after last commit in args
* Handle not allowed methods according to HTTP/1.1 (RFC 7231) https://www.rfc-editor.org/rfc/rfc7231

isoos commented on CHANGELOG.md:
```diff
@@ -11,7 +17,7 @@
 ## 1.4.2
 
-- Support POST method in `CurlClient`. ([#9](https://github.com/isoos/http_client/pull/9) by [rbandres98](https://github.com/rbandres98))
+- Support POST method in `CurlClient`. ([#9](https://github.com/isoos/http_client/pull/9) by [andresruizdev](https://github.com/andresruizdev))
```
I think you wanted to copy this block up to 1.5.2, but instead edited it in-place.

andresruizdev replied:
> I think you wanted to copy this block up to 1.5.2, but instead edited it in-place.

rbandres98 is my old username, and currently CHANGELOG.md is redirecting to the profile https://github.com/rbandres98, a profile that no longer exists since I updated my username to andresruizdev (https://github.com/andresruizdev). So I just updated the changelog with the newer username and url.

isoos replied:
Oh, alright. Still, please add the same note to the 1.5.2 entry.

andresruizdev replied:
> Oh, alright. Still, please add the same note to the 1.5.2 entry.

Ok, thanks for the suggestion, I modified the CHANGELOG.md file with that information.

andresruizdev replied:
I forgot to upload the pull request reference in the changelog. I already added it too, sorry.

isoos reviewed (changes requested):
On request in the changelog, otherwise looks good, thanks!

andresruizdev committed:
Update rbandres98 to andresruizdev

rbandres98 was the previous username, and the CHANGELOG.md was redirecting to the profile https://github.com/rbandres98, which no longer exists as the username has been updated to andresruizdev (https://github.com/andresruizdev). The CHANGELOG.md has now been updated with the new username and URL.

andresruizdev committed:
Add PR reference

Add reference to PR #14

isoos commented:
published
